operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java

```diff
@@ -98,8 +104,10 @@ public void createOrUpdate(ManagedKafka managedKafka) {
         Secret currentSsoTlsSecret = cachedSecret(managedKafka, ssoTlsSecretName(managedKafka));
 
         if (isKafkaAuthenticationEnabled(managedKafka)) {
+            Secret ssoClientSecret = ssoClientSecretFrom(managedKafka, currentSsoClientSecret, currentMasterSecret);
+            createOrUpdate(ssoClientSecret);
```

Comment:
I think something with this part needs to change. Perhaps have `ssoClientSecretFrom` also check that `managedKafka.getSpec().getOauth().getClientSecret()` is not null before using it. Otherwise return `null` and call `createOrUpdate` only when a nonnull value was returned. 

The check in `AbstractKafkaCluster#buildSsoClientGenericSecretSource` should also check that the clientSecretRef is null before returning null.